src/routers/S3StorageAPI/S3StorageAPI.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-env node */
+'use strict';
+
+const express = require('express'),
+    router = express.Router(),
+    bodyParser = require('body-parser'),
+    Minio = require('minio');
+
+/* eslint-disable max-lines-per-function */
+function initialize(middlewareOpts) {
+    const logger = middlewareOpts.logger.fork('S3StorageAPI'),
+        ensureAuthenticated = middlewareOpts.ensureAuthenticated;
+    let client;
```

Comment:
I would declare this in each method rather than relying on a shared variable in async methods. This can cause difficult to catch errors when there is higher usage of this endpoint by multiple users/buckets.

Specifically, in the current code, suppose there are two incoming requests that are almost simultaneous: a request to `/createBucket` and one to `/statObject`. Suppose the request to `/createBucket` arrives first and each request is operating on a completely different s3 endpoint. The handler for `/createBucket` will execute first and then yield control of the process once it hits 
```javascript
            resObj.alreadyExists = await client.bucketExists(req.body.bucketName);
```
At this time, the handler for `/statObject` will be invoked. This will change the value of the *shared* `client` variable to be configured for the request to `/statObject` (potentially another s3 url entirely!). When the handler for bucket creation resumes, the following line will actually create the bucket in the context of the `/statObject` request rather than in the desired location:
```javascript
                await client.makeBucket(req.body.bucketName);
```

Reply:
Instead of using a shared variable, each route now uses a separate variable.